### PR TITLE
load cmd support gzip

### PR DESCRIPTION
**Pull request by Junnplus**
Fixes https://github.com/containerd/nerdctl/issues/581

### CODE REVIEW

File: cmd/nerdctl/load.go [hunk 1]
```diff
@@ -68,6 +72,16 @@ func loadAction(cmd *cobra.Command, args []string) error {
...
 
+	inbuf := bufio.NewReader(in)
+	buff, _ := inbuf.Peek(3)
+	if bytes.HasPrefix(buff, []byte("\x1f\x8b")) {
+		in, _ = gzip.NewReader(inbuf)
+	} else if bytes.HasPrefix(buff, []byte("\x42\x5a\x68")) {
+		in = bzip2.NewReader(inbuf)
+	} else {
+		in = inbuf
+	}
```
Comment (AkihiroSuda) [inline cmd/nerdctl/load.go:83]: Thanks for working on this, but I'd suggest using https://pkg.go.dev/github.com/containerd/containerd/archive/compression#DecompressStream

Comment (Junnplus) [reply cmd/nerdctl/load.go:83]: `DecompressStream` only supports gzip and zstd.
https://github.com/containerd/containerd/blob/main/archive/compression/compression.go#L169

but `docker load` support gzip and bzip2.
https://docs.docker.com/engine/reference/commandline/load/#extended-description

Comment (AkihiroSuda) [reply cmd/nerdctl/load.go:83]: Then we should fix containerd/containerd

Comment (AkihiroSuda) [reply cmd/nerdctl/load.go:83]: Anyway, supporting bzip2 is probably not important. Is anyone really using bzip2 for Docker v1 archives?

Comment (Junnplus) [reply cmd/nerdctl/load.go:75]: @AkihiroSuda I first change to  using`DecompressStream`,

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**